text/0000-dynamic-import-hints.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Dynamic Import Hints and Its Metadata
+status: DRAFTED
+created_at: 2020-03-16
+updated_at: 2020-03-22
+rfc: https://github.com/salesforce/lwc-rfcs/pull/27
+champion: Aliaksandr Papko (@apapko) | LWC Team
+implementation: https://git.soma.salesforce.com/lwc/lwc-api/pull/42 & https://git.soma.salesforce.com/lwc/lwc-platform/pull/361
+---
+
+# Hints Syntax and Its Metadata
+
+## Summary
+This RFC suggests authoring syntax for specifying dynamic import hints and describes its metadata shape.
+
+## Motivation
+Using hints allow component authors to declare the component preload context by parameterizing dynamic imports. The values from the hints can be used to optimize dependency resolution and pre-fetching.
```

Comment:
> Using hints allow component authors to declare the component preload context by parameterizing dynamic imports.

This sentence feels a little cryptic. What about something along the lines of:

"Using hints allows component authors to specify a condition on which a dynamic import will be necessary. This is a way to describe the conditions required for this import to happen. An application framework can use these hints to optimize the loading of those resources (for example: pre-fetch at the time of resolving dependencies)"

As I said below an example showing an if statement might be a little more clear.